js/modules/k6/experimental/fs/module.go

```diff
@@ -209,15 +232,28 @@ func (f *File) Read(into goja.Value) *goja.Promise {
 		return promise
 	}
 
-	// Obtain the underlying byte slice from the ArrayBuffer.
-	// Note that this is not a copy, and will be modified by the Read operation
-	// in place.
-	buffer := ab.Bytes()
+	// Copy the ArrayBuffer into a byte slice, so that we can pass it to the
+	// [file.Read] method without risking to modify the original ArrayBuffer, and
+	// running into concurrency issues (data race).
+	intoBytes := ab.Bytes()
+	buffer := make([]byte, len(intoBytes))
+	_ = copy(buffer, intoBytes)
```

Comment:
Is there a reason for this `copy` ?

It seems to me like there is no guarantee that `Read` will keep the bytes in the buffer it doesn't read. 

> Even if Read returns n < len(p), it may use all of p as scratch space during the call

From [Reader](https://pkg.go.dev/io#Reader)

And even if it was - you already copy on the way back.

So it seems like we here only want to make a buffer with the correct length

Reply:
The idea there was that we save the state of the input buffer, as given by the runtime, before starting the async operation of reading into it, so that we read (and return) into a buffer that's in a consistent state with the one we receive as input.

In light our internal discussions, I'm not sure that's the correct approach indeed, and this `copy` can probably go 👍🏻